client/src/components/Framework/Framework.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import AceEditor from 'react-ace';
+import 'brace/mode/html';
+import 'brace/mode/css';
+import './css/Cards.css';
+import './css/Framework.css';
+import blackCards from './black-cards.js'
+import whiteCards from './white-cards.js'
+import { whiteHTML, whiteCSS, blackHTML, blackCSS } from './data/markup-strings.js'
+
+export default class Framework extends React.Component {
+  constructor(props){
+		super(props)
+		this.state = {
+			blackCard: null,
+			whiteCard: null
+		}
+  }
+
+  componentDidMount(){
+		document.body.classList.toggle('dark-body', this.props.isDark)
+		this.changeBlackBody();
+		this.changeWhiteBody();
+	}
+	
+	changeBlackBody(){
+		let ranBlackCard = blackCards[Math.floor(Math.random() * blackCards.length)];
+		this.setState({
+			blackCard: ranBlackCard.text
+		})
+	}
+
+	changeWhiteBody(){
+		let ranWhiteCard = whiteCards[Math.floor(Math.random() * whiteCards.length)];
+		this.setState({
+			whiteCard: ranWhiteCard
+		})
+	}
+
+  render(){
+    return (
+			<div className="container">
+				<div className="header-body">
+					This is the framework playground and style guide for Cards Against Corndel.<br/>
+					<h5>Reference material used for designing the styling for CAC.</h5>
+				</div>
+
+				<div className="section">
+					<div className="white-card">
+						<div className="white-card-body" dangerouslySetInnerHTML={{__html: this.state.whiteCard}}>
+						</div>
+					</div>
+					<div className="section-side">
+						<h6 className="section-header">HTML</h6>
+						<AceEditor
+							className="editor"
+							mode="html"
+							theme="github"
+							name="code-editor"
+							width="300px"
+							height="300px"
+							value={whiteHTML}
+							maxLines="4"
+							setOptions={{readOnly: true}}
+						/>
+						<h6 className="section-header">CSS</h6>
+						<AceEditor
+							className="editor"
+							mode="css"
+							theme="github"
+							name="code-editor"
+							width="300px"
+							height="300px"
+							value={whiteCSS}
+							maxLines="13"
+							setOptions={{readOnly: true}}
+						/>
+						<button className="btn-changer" onClick={() => this.changeWhiteBody()} type="submit">Test Body</button>
+					</div>
+				</div>
+
+				<div className="section">
```

Comment:
This black and white card business is quite duplicated code, I wonder if it's possible to commonise it by having a function thats like `renderCard(card, black/white)`

Reply:
I'm not sure if that's better or not, thoughts?

Reply:
I'm looking at lines 26-38 and wondering if we can add to this - if the way we're going to add cards is to give them a state 'indeck/discarded/inplay' can we do this when they're set? So they have a 'blackcard' state and a 'blackcardstatus' state, which defaults to 'in deck', changes to 'inplay' when it's drawn and when is played changes to 'discarded'?

But I do agree with Josh on the duplication of code. Could start with the function with an 'if white, else black' and then whittle it down more?

Reply:
I think this would be sensible when the cards are generated in the game itself, the code from 26-38 is just there to display the different bodies that the card might have. I was just using it as a test to see how the styling of the card behaved dependant on how much text was present in the body. When the cards are generated in the actual game, maybe we could recycle some of this code and implement your logic?